tests/test_plugin.py

```diff
@@ -15,39 +15,81 @@
 
 
 @pytest.mark.parametrize("directory_urls", ["use_directory_urls", "no_directory_urls"])
-@pytest.mark.parametrize("nav", ["explicit_nav", "derived_nav"])
-def test_real_example(tmpdir, directory_urls, nav):
+@pytest.mark.parametrize("nav_src", ["explicit_nav", "derived_nav"])
+@pytest.mark.parametrize("index_file", ["default_index_file", "index.md", "foo.md"])
+def test_real_example(tmpdir, directory_urls, nav_src, index_file):
     config = dict(
         docs_dir=str(example_dir / "docs"),
         site_dir=tmpdir,
         use_directory_urls=(directory_urls == "use_directory_urls"),
-        nav=load_config(str(example_dir / "mkdocs.yml"))["nav"] if nav == "explicit_nav" else None,
+        nav=load_config(str(example_dir / "mkdocs.yml"))["nav"]
+        if nav_src == "explicit_nav"
+        else None,
     )
+    if nav_src == "derived_nav" and index_file != "default_index_file":
```

Comment:
Instead of adding actual files to the real example and branching all of this logic, I think the test should be separate and new.

If you need to use files from the real example, you can first create a directory for it then copy those files and add more.

```python
shutil.copytree(example_dir / "docs", str(tmpdir))

noindex_dir = (tmpdir / "docs").mkdir("z_noindex")
(noindex_dir / "a.md").write_text("# a")
```

then change `docs_dir=tmp_dir,`


Or if you don't need the real example, just don't copy the files

Reply:
Sounds good